### PR TITLE
[Cong Le] Implement add competency

### PR DESCRIPTION
**Pull request by ltc2310**

### CODE REVIEW

**Comment by ltc2310** [comment]
[CF] Add new competency and update activate competency.
Add new competency : It doesn't update page in real time
Update activate competency : It works but don't update page and onChange function when I click in the first time it doesn't work,on second time it works but the value opposite with data in firebase

File: src/Competencies.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
+import React, {Component} from 'react';
+import { Modal,Switch, Icon, Button,Input } from 'antd';
+import {getData,insert,listenData} from './firebase';
+import _ from 'lodash';
+import Loading from './Loading';
+import './competency.css';
+import {Router, Link} from 'react-router-component';
+
+var list;
```
Comment (hunterbmt) [inline src/Competencies.js]: Do not use "var" in ES6

File: src/Competencies.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+          </div>
+      </div>
+    );
+  }
+}
+
+class CompetencyList extends Component {
+  constructor(props) {
+     super(props);
+     list=this;
```
Comment (hunterbmt) [inline src/Competencies.js]: Why do we need to assign this to outside variable?

File: src/Competencies.js [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                this.state.arrayCompetencies.map((competency,index) => {
+                return <Competency key={index}  name={(competency)} />
+              })
+            }
+        </div>
+      );
+    }
+}
+
+class CompetencyPopup extends Component {
```
Comment (hunterbmt) [inline src/Competencies.js]: one component per file

File: src/Competencies.js [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+  handleChange(e){
+     this.setState({competencyName: e.target.value});
+  }
+
+  handleSave=(e) => {
+    this.setState({
+      visible: false
+    });
+
+    console.log(list.state.arrayCompetencies);
```
Comment (hunterbmt) [inline src/Competencies.js]: Removing console.log

File: src/Competencies.js [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+    this.addNewCompetency(this.state.competencyName);
+
+    var newArrayCompetencies = list.state.arrayCompetencies.push(this.state.competencyName);
```
Comment (hunterbmt) [inline src/Competencies.js]: Use this instead of outside variable

**Review by hunterbmt — CHANGES_REQUESTED** [review]

**Comment by ltc2310** [comment]
Fix comment and rewrite function follow new database
How to set auto increment key on competencies when I save new competency?
When I get data source for competencies KMScore to show on table,It return double data competency show on table?

**Comment by ltc2310** [comment]
* [CFT] fix add competency follow new database but it has some issues
- I can't set auto increment when I save new competency
- I can't load two table KMS Core and KMS optional in the same time
- When I click on button it always append data on the table

File: src/Competencies.js [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+
+  }
+
+  componentDidMount() {
+
+
+  }
+
+  getDataSoureKMSCore() {
+    _.map(this.state.competenciesKMSCore, (item, index) => {
```
Comment (hunterbmt) [inline src/Competencies.js]: const dataSourceCore = _.map(() => return {}) ;
this.setState({dataSourceCode});

**Pull request closed without merging**